Source/astcenccli_toplevel.cpp

```diff
@@ -71,6 +71,28 @@ static uint32_t unpack_bytes(
 	       ((uint32_t)(d) << 24);
 }
 
+enum astc_op_mode {
+	ASTC_ENCODE,
+	ASTC_DECODE,
+	ASTC_ENCODE_AND_DECODE,
+	ASTC_PRINT_LONGHELP,
+	ASTC_PRINT_VERSION,
+	ASTC_UNRECOGNIZED
+};
+
+astcenc_swizzle swz_encode {
```

Comment:
I'd prefer both swizzle structures to be locals in main() which are passed around; in general try to avoid adding new globals if we can.

Reply:
done